core/templates/dev/head/pages/exploration_editor/ExplorationStatesService.js

```diff
@@ -402,16 +402,25 @@ oppia.factory('ExplorationStatesService', [
         $rootScope.$broadcast('refreshGraph');
       },
       registerOnStateAddedCallback: function(callback) {
-        stateAddedCallbacks.push(callback);
+        if (stateAddedCallbacks.indexOf(callback) === -1) {
```

Comment:
I feel rather iffy about introducing the pattern of using indexOf with functions. When are two functions equal? That's not clear to me. Must their names be equal, or their function bodies, or their args, or their contexts, or some combination of these?

I can see two alternative ways around this. You could have a setOnStateAddedCallback and make the corresponding variable a function|null (rather than an array). Or alternatively you could have this register method have "append" connotations, and have a separate "clear" method. I think either of these would be better than having "register" mean "append if doesn't already exist".

Ditto below.

Reply:
Done.